compiler/qsc/src/interpret.rs

```diff
@@ -656,29 +728,46 @@ impl Interpreter {
         entry: CircuitEntryPoint,
         simulate: bool,
     ) -> std::result::Result<Circuit, Vec<Error>> {
-        let entry_expr = match entry {
+        let (entry_expr, invoke_params) = match entry {
             CircuitEntryPoint::Operation(operation_expr) => {
                 let (item, functor_app) = self.eval_to_operation(&operation_expr)?;
                 let expr = entry_expr_for_qubit_operation(item, functor_app, &operation_expr)
                     .map_err(|e| vec![e.into()])?;
-                Some(expr)
+                (Some(expr), None)
             }
-            CircuitEntryPoint::EntryExpr(expr) => Some(expr),
-            CircuitEntryPoint::EntryPoint => None,
+            CircuitEntryPoint::EntryExpr(expr) => (Some(expr), None),
+            CircuitEntryPoint::Callable(call_val, args_val) => (None, Some((call_val, args_val))),
+            CircuitEntryPoint::EntryPoint => (None, None),
         };
 
         let circuit = if simulate {
             let mut sim = sim_circuit_backend();
 
-            self.run_with_sim_no_output(entry_expr, &mut sim)?;
+            match invoke_params {
+                Some((callable, args)) => {
+                    let mut sink = std::io::sink();
```

Comment:
how does this `std::io` sink work here when running in, e.g., wasm? I'm sure it works, I'm just used to us running all output through our logging abstractions

Reply:
I think it just pipes stuff to null in an efficient way... or as [the docs](https://doc.rust-lang.org/std/io/struct.Sink.html) put it:

> A writer which will move data into the void.

We already used this pattern for a couple of other places, notably the resource estimation path, so I think it's considered safe.